tsl/src/compression/compress_utils.c

```diff
@@ -420,7 +420,7 @@ tsl_compress_chunk_wrapper(Chunk *chunk, bool if_not_compressed)
 		ereport((if_not_compressed ? NOTICE : ERROR),
 				(errcode(ERRCODE_DUPLICATE_OBJECT),
 				 errmsg("chunk \"%s\" is already compressed", get_rel_name(chunk->table_id))));
-		return false;
+		return if_not_compressed ? true : false;
```

Comment:
Could you add a function comment documenting the expectations on the return type and why. Otherwise there is a risk that this is accidentally changed later.